File: app/src/main/java/catalina/joshua/calcularcuota/MainActivity.java
package catalina.joshua.calcularcuota;

import android.content.Context;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import org.w3c.dom.Text;

import java.text.NumberFormat;
import java.util.ArrayList;

import static android.R.attr.duration;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        float valor_cuota = 0.0f;

        Button btn_calcular = (Button) findViewById(R.id.btn_calcular);
        final TextView txt_monto = (TextView) findViewById(R.id.input_monto);
        final TextView txt_interes = (TextView) findViewById(R.id.input_interes);
        final TextView txt_plazo = (TextView) findViewById(R.id.input_cuotas);
        final TextView txt_valor_cuota = (TextView) findViewById(R.id.valor_cuota);

        btn_calcular.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                float monto = Float.parseFloat( txt_monto.getText().toString() );
                float interes = Float.parseFloat( txt_interes.getText().toString() );
                double _R = 0;
                int plazo = Integer.parseInt(txt_plazo.getText().toString());
                ArrayList<Cuota> plan_pagos = new ArrayList<Cuota>();
                interes = interes / 100;
                _R = monto / (( 1 -  Math.pow((1 + interes), (-plazo))) / interes );

                if (interes == 0.0){
                    showMensaje(MainActivity.this.getApplicationContext(), "El interes no puede ser cero", Toast.LENGTH_LONG);
                }
                if (monto == 0.0){
                    showMensaje(MainActivity.this.getApplicationContext(), "El monto no puede ser cero", Toast.LENGTH_LONG);
                }
                if (plazo == 0){
                    plazo = 1;
                }

                for (int i=0; i< plazo; i++){
                    Cuota c = new Cuota();
                    if (i > 0){
                        c.setPago(_R);
                        if (interes == 1){
                            c.setCapital(monto);
                            c.setInteres(0);
                            c.setPago(monto);
                            c.setBalance(0);
                        }else{
                            c.setInteres(plan_pagos.get(i-1).getBalance() * interes);
                            c.setCapital(c.getPago() - c.getInteres());
                            c.setBalance(c.getBalance() - c.getCapital());
                        }
                    }else{
                        c.setBalance(monto);
                    }

                    plan_pagos.add(c);

                }
                Cuota respuesta = plan_pagos.get(1);
                NumberFormat format = NumberFormat.getCurrencyInstance();
                txt_valor_cuota.setText( format.format( respuesta.getPago() ) );

            }
        });
    }

    private void showMensaje(Context context, String texto, int duracion){
        Toast toast = Toast.makeText(context, texto, duracion);
        toast.show();
    }
}

File: app/src/main/java/com/example/calculocuota/MainActivity.kt
package com.example.calculocuota

import android.graphics.Color
import android.os.Bundle
import android.view.ViewGroup
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.setPadding
import androidx.core.widget.doAfterTextChanged
import com.example.calculocuota.dto.Cuota
import java.text.DecimalFormat


class MainActivity : AppCompatActivity() {
    var planPagos: MutableList<Cuota> = ArrayList<Cuota>()
    val dec = DecimalFormat("###,###.##")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val monto = findViewById<EditText>(R.id.txtMonto)
        val plazo = findViewById<EditText>(R.id.txtPlazo)
        val cuota = findViewById<TextView>(R.id.resultado)
        val interes = findViewById<EditText>(R.id.txtInteres)
        val boton = findViewById<Button>(R.id.btnCalcular)
        val tablaCuotas = findViewById<TableLayout>(R.id.tablaCuotas)

        boton.setOnClickListener{
            this.planPagos.clear()
            val valorMonto = monto.text.toString().replace(",","")
            val plazoValor = plazo.text.toString()
            val interesValor = interes.text.toString()
            cuota.text = dec.format( calcularCuota(valorMonto,plazoValor, interesValor) )
            pintarTabla(tablaCuotas)
        }
    }

    fun pintarTabla(tabla:TableLayout){
        tabla.removeAllViews()
        var id = 1
        this.planPagos.forEach{ cuota ->

            var _tr = TableRow(this)
            _tr.id = id
            _tr.setBackgroundColor(Color.LTGRAY)
            _tr.layoutParams = ViewGroup.LayoutParams(TableRow.LayoutParams.MATCH_PARENT,TableRow.LayoutParams.WRAP_CONTENT)
            var _cuotaNumero = TextView(this)
            _cuotaNumero.id = id + 100
            _cuotaNumero.text = id.toString()
            _cuotaNumero.setPadding(5,5,5,5)
            var _cuota = TextView(this)
            _cuota.id = id + 200
            _cuota.text = dec.format( cuota.monto)
            _cuota.setPadding(5,5,5,5)
            _tr.addView(_cuotaNumero)
            _tr.addView(_cuota)
            tabla.addView(_tr)
            id++
        }
    }

    fun calcularCuota(_monto:String, _plazo:String, _interes:String): Float {
        val monto = _monto.toFloat()
        val plazo = Integer.parseInt(_plazo)

        if (_plazo == "1"){
            return (monto/plazo)
        }

        val interesProcentaje = _interes.toFloat()
        var sumatoria = 0.0F
        var montoPlazo = monto

        for (i in 1 .. plazo){
            var interes = montoPlazo * (interesProcentaje/100)
            var cuota = (monto / plazo) + interes
            if (i == 1){
                cuota = montoPlazo / plazo
                montoPlazo = montoPlazo - cuota
            }
            if (i == 2){
                cuota = cuota + planPagos.get(0).interes
                montoPlazo = montoPlazo - (cuota -  planPagos.get(0).interes - interes)
            }
            if (i > 2) {
                montoPlazo = montoPlazo - (cuota - interes)
            }
            var cuotaObj = Cuota()
            cuotaObj.interes = interes
            cuotaObj.monto = cuota
            cuotaObj.deudaTotal = montoPlazo
            planPagos.add(cuotaObj)
            sumatoria += cuota
        }

        var promedio = sumatoria / plazo
        return promedio
    }

}